tutorials/tour/basics.md

```diff
@@ -0,0 +1,207 @@
+---
+layout: tutorial
+title: Basics
+
+disqus: true
+
+tutorial: scala-tour
+num: 2
+next-page: unified-types
+previous-page: tour-of-scala
+---
+
+In this page, we will cover basics of Scala.
+
+## Variables
+
+You can define variables with `var` keyword.
+
+```
+var x = 1 + 1
+x += 1
+```
+
+Often, you want your variables to be immutable. You can use `val` keyword in that case.
+
+```
+val x = 1 + 1
+x += 1 // This does not compile because you declared the variable with "val" keyword
+```
+
+Type of variables can be inferred, but you can also explicitly state type like below.
+
+```
+val x: Int = 1 + 1
+```
+
+## Functions
+
+You can define functions like below.
+
+```
+(name: String) => println("Hello, " + name + "!")
+```
+
+You can also assign functions to variables.
+
+```
+val greet = (name: String) => println("Hello, " + name + "!")
+greet("Scala") // Hello, Scala!
+```
+
+We will cover functions in depth [later](anonymous-function-syntax.md).
+
+## Classes
+
+You can define classes with `class` keyword.
+
+```
+class Point(var x: Int, var y: Int) {
+  def move(dx: Int, dy: Int): Unit = {
+    x = x + dx
+    y = y + dy
+  }
+
+  override def toString: String = "(" + x + ", " + y + ")"
+}
+```
+
+You can instantiate classes with `new` keyword.
+
+```
+val point = new Point(1, 2)
+println(point) // (1, 2)
+```
+
+We will cover classes in depth [later](classes.md).
+
+## Case Classes
+
+By default, classes are compared by reference. If you want to compare them by values, you need to override `equals` and `hashCode` methods.
```

Comment:
I would enumerate more clearly the benefits of a case class over a class, either here or at the end of this section. I think it's good to show them need by need, but they should be summarized somewhere :smile:.